src/Compilers/CSharp/Portable/Lowering/ClosureConversion/ClosureConversion.cs

```diff
@@ -622,7 +622,14 @@ protected override BoundExpression FramePointer(SyntaxNode syntax, NamedTypeSymb
                 // However, frame pointer local variables themselves can be "captured".  In that case
                 // the inner frames contain pointers to the enclosing frames.  That is, nested
                 // frame pointers are organized in a linked list.
-                return proxyField.Replacement(syntax, frameType => FramePointer(syntax, frameType));
+                return proxyField.Replacement(
+                    syntax,
+                    static (frameType, arg) =>
+                    {
+                        var (syntax, @this) = arg;
+                        return @this.FramePointer(syntax, frameType);
+                    },
+                    (syntax, this));
```

Comment:
It seems like many of these could be significantly more succinct. Feels like we're "paying more than we need to" for the gains.

```suggestion
                    static (frameType, arg) => arg.self.FramePointer(arg.syntax, frameType),
                    (syntax, self: this));
```

Reply:
Sure, I can switch to that form.